core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -225,14 +242,33 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     } yield ()
   }
 
-  def deleteBillingProject(projectName: RawlsBillingProjectName, ownerInfo: Map[String, String]): Future[PerRequestMessage] = {
+  def adminDeleteBillingProject(projectName: RawlsBillingProjectName, ownerInfo: Map[String, String]): Future[PerRequestMessage] = {
     // unregister then delete actual project in google
+    val ownerUserInfo = UserInfo(RawlsUserEmail(ownerInfo("newOwnerEmail")), OAuth2BearerToken(ownerInfo("newOwnerToken")), 3600, RawlsUserSubjectId("0"))
     for {
-      _ <- unregisterBillingProject(projectName, ownerInfo)
+      _ <- unregisterBillingProjectWithUserInfo(projectName, ownerUserInfo)
       _ <- gcsDAO.deleteProject(projectName)
     } yield RequestComplete(StatusCodes.NoContent)
   }
 
+  def deleteBillingProject(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = {
+    for {
+      // Check if workspace exists for this project.
+     workspaces <- dataSource.inTransaction { dataAccess =>
+        dataAccess.workspaceQuery.listByNamespace(projectName)
+      }
+
+     _<- if(!workspaces.isEmpty) {
+       Future.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, "There are workspaces exists in this project.")))
```

Comment:
I'm not sure that this is the right status code to return in this case. My mental model for 403 Forbidden is when the user doesn't have the correct permissions to perform an action. In this case, the user has the correct permissions, but can't delete the billing project because there are associated workspaces. Maybe a 400 Bad Request would be appropriate here? I'm definitely open to keeping it as a 403 if you think that's better though

Reply:
https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala#L96
Is the error where it comes from

Reply:
Sure, but in this case `requireProjectAction` returned true, but we are now rejecting the request because there's still a workspace in the project. The reason the call to delete the billing project is failing isn't because the caller is lacking permissions, rather it's because the caller hasn't yet deleted all the necessary billing projects

Reply:
Ah, sorry, I thought you are asking the line of error when user doesn't have permission. For workspace already exists, I also think 400 is better. Thanks!